### ISSUE TITLE
compute_external_vpn_gateway lacks resource labels

### ISSUE THREAD

**Issue by trodge** [hashicorp/terraform-provider-google#7798]
The TF provider implementation for compute_external_vpn_gateway does not have a field supporting resource labels. Is there a reason why some resources support labels and not others? Labels exist in the underlying REST API representation of the resource, but cannot be set using TF provider.

**Comment by edwardmedia** [comment]
`label` is supported in API (below). Added labels to `enhancement` for triage

https://cloud.google.com/compute/docs/reference/rest/v1/externalVpnGateways

**Comment by AlfatahB** [comment]
b/263841783

**Comment by AlfatahB** [comment]
I've looked into this issue and following are the details I found out:
- It seems like we need to support the `labels` field for the `google_compute_external_vpn_gateway` resource.
- `labels` field can be found in the [API of the externalVpnGateway resource](https://cloud.google.com/compute/docs/reference/rest/v1/externalVpnGateways#resource:-externalvpngateway).
- As this resource is MMv1 generated, we need to add this field using the `api.yaml` file for the `compute` product.
